### PR TITLE
[shared_preferences] Documentation update for Readme.md

### PR DESCRIPTION
**Pull request by jpsiitonen**
Update for documentation on readme file:
- Improved documentation details about plugin
- Added short API usage examples for shared preferences API usage. 

*List which issues are fixed by this PR. You must list at least one issue.*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by alexlindroos** [comment]
LGTM

File: packages/shared_preferences/shared_preferences/README.md [hunk 1]
```diff
@@ -7,10 +7,65 @@ Wraps platform-specific persistent storage for simple data
 and there is no guarantee that writes will be persisted to disk after
 returning, so this plugin must not be used for storing critical data.
 
+Although key-value storage is easy and convenient to use, it has limitations:
+* Only primitive types can be used: `int`, `double`, `bool`, `string`, and `stringList`.
```
Comment (tomgilder) [inline packages/shared_preferences/shared_preferences/README.md:11]: If these refer to Dart types:

```suggestion
* Only primitive types can be used: `int`, `double`, `bool`, `String`, and `StringList`.
```

File: packages/shared_preferences/shared_preferences/README.md [hunk 2]
```diff
@@ -7,10 +7,65 @@ Wraps platform-specific persistent storage for simple data
...
+* It’s not designed to store a lot of data.
```
Comment (tomgilder) [inline packages/shared_preferences/shared_preferences/README.md:12]: Should we mention that it's not encrypted too?

File: packages/shared_preferences/shared_preferences/README.md [hunk 3]
```diff
@@ -7,10 +7,65 @@ Wraps platform-specific persistent storage for simple data
...
+
+SharedPreferences data is deleted in these situations:
+* on uninstallation of app or
+* on clearing the application data.
```
Comment (tomgilder) [inline packages/shared_preferences/shared_preferences/README.md:16]: Is this just on Android, as it refers to `SharedPreferences`? Can probably just be one sentence, not a list.

File: packages/shared_preferences/shared_preferences/README.md [hunk 4]
```diff
@@ -7,10 +7,65 @@ Wraps platform-specific persistent storage for simple data
...
 
-### Example
+Add the following import: 
+```dart
+import 'package:shared_preferences/shared_preferences.dart';
+```
+
+### Examples
+Here are small examples that show you how to use the API. 
+Please also take a look at our [example][example] app.
```
Comment (tomgilder) [inline packages/shared_preferences/shared_preferences/README.md:28]: I don't think the example link will work when this is published to pub.dev.

Might be better just getting rid of this paragraph.

Comment (jpsiitonen) [reply packages/shared_preferences/shared_preferences/README.md:28]: It should work , at least other plugins seems to work fine with links. But true, probably no need for this in readme.

File: packages/shared_preferences/shared_preferences/README.md (hunk 3 above)
Comment (jpsiitonen) [reply packages/shared_preferences/shared_preferences/README.md:16]: reformatted to text, basically yes happens on android - not sure about all the platforms

**Pull request closed without merging**